### PR TITLE
Add ceil and floor built-ins

### LINKED ISSUE #2930: Add ceil() function
We have `round()` function at the moment. But it would be nice to have ceil() and floor() math functions as well.

```
# 1.0 -> 1.0
ceil(value) = result {
  round(value) == value
  result := value
}

# 1.6 -> 2
ceil(value) = result {
  round(value) - value > 0
  result := round(value)
}

# 1.1 -> 2
ceil(value) = result {
  (round(value) - value) < 0
  result := round(value) + 1
}
```

Thanks!

### PR DESCRIPTION
**Pull request by anderseknert**
Closes #2930

Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: docs/content/policy-reference.md [hunk 1]
```diff
@@ -272,6 +272,8 @@ complex types.
 | <span class="opa-keep-it-together">``z := x / y``</span>   | ``z`` is the quotient of ``x`` and ``y``  | ✅ |
 | <span class="opa-keep-it-together">``z := x % y``</span>   | ``z`` is the remainder from the division of ``x`` and ``y``  | ✅ |
 | <span class="opa-keep-it-together">``output := round(x)``</span>    | ``output`` is ``x`` rounded to the nearest integer | ✅ |
+| <span class="opa-keep-it-together">``output := ceil(x)``</span>    | ``output`` is ``x`` rounded up to the nearest integer | ✅ |
+| <span class="opa-keep-it-together">``output := floor(x)``</span>    | ``output`` is ``x`` rounded down the nearest integer | ✅ |
```
Comment (anderseknert) [inline docs/content/policy-reference.md:276]: Note that the WASM support isn't actually there yet - I'm hoping someone can help me with that before next release though ;) 

CC @srenatus @koponen-styra

Comment (srenatus) [reply docs/content/policy-reference.md:276]: Sure thing. But let's keep the docs inline with the code base? 😉 We can tackle these in follow-up PRs, no need to block this.

Comment (anderseknert) [reply docs/content/policy-reference.md:276]: Sure, I can change this - I take it "SDK-dependent" is the politically correct way of saying "No"? 😄 I don't see any other option listed in the docs for any function.

**Comment by srenatus** [comment]
🔍 So in the generated WASM module, we'll end up generating a call that's invoking [this `ceil()`](https://github.com/open-policy-agent/opa/blob/a3aa77779a4135cf25536a96b184cfe7e2d387b8/wasm/src/lib/math.h#L11) and not the builtin. 

I think the best way forward would be to fix that by implementing it in our WASM native builtins, I'll give it a try.

File: docs/content/policy-reference.md (hunk 1 above)
Comment (srenatus) [reply docs/content/policy-reference.md:276]: Not a lie anymore 🎉

**Review by srenatus — APPROVED** [review]
I approve of what I haven't authored 😉

File: wasm/src/arithmetic.c [hunk 2]
```diff
@@ -50,6 +50,50 @@ opa_value *opa_arith_round(opa_value *v)
...
+
+    mpd_t *r = mpd_qnew();
+    uint32_t status = 0;
+
+    mpd_qfloor(r, n, mpd_max_ctx(), &status);
+    mpd_del(n);
+
+    if (status)
+    {
+        return NULL;
```
Comment (ashutosh-narkar) [inline wasm/src/arithmetic.c:91]: Nit: Can we add a test case for this ? Also some of the other functions abort on non-zero status. Is there any reason for different behavior here ?

Comment (srenatus) [reply wasm/src/arithmetic.c:91]: Yeah sorry, should have called that out! 😅 This is in line with the changes done to the other arithmetic builtins in #3022, I think.

File: test/cases/testdata/arithmetic/test-arithmetic-0824.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+cases:
+- data: {}
+  modules:
+  - |
+    package generated
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply test/cases/testdata/arithmetic/test-arithmetic-0824.yaml]: OK, done.

**Comment by ashutosh-narkar** [comment]
@anderseknert wasn't there a commit which had the Wasm implementation ?

**Comment by anderseknert** [comment]
Ah, crap, I rebased from master but forgot to pull @srenatus commit before pushing 🤦  Stephan, can you re-add your commit on top?

**Comment by srenatus** [comment]
rebased and f-pushed, too. ✔️

**Review by ashutosh-narkar — APPROVED** [review]
LGTM

**Review by tsandall — APPROVED** [review]
LGTM
